src/addrman.h

```diff
@@ -27,7 +27,11 @@ class InvalidAddrManVersionError : public std::ios_base::failure
 class AddrManImpl;
 
 /** Default for -checkaddrman */
+#ifdef DEBUG_ADDRMAN
+static constexpr int32_t DEFAULT_ADDRMAN_CONSISTENCY_CHECKS{1};
```

Comment:
If these checks are going to be enabled by default on some builds, shouldn't the default be much higher; eg 100 or 1000? Doing it on every call is useful for live debugging or small scale tests, but seems overkill for a node's runtime default?

Reply:
Yes I agree, the performance hit is too much for any kind of normal operation.

I updated the branch with @MarcoFalke's suggestion along with a new debug option (which I called `--enable-debug-checks`, and not `--debug-slow`).

I'll run a few runtime tests on various values for the checks to see if I can find a sweet spot for the frequency, then update the doc commit and push an update here.